soc/arm/ti_simplelink/Kconfig

```diff
@@ -4,7 +4,6 @@
 
 config SOC_FAMILY_TISIMPLELINK
 	bool
-	select LEGACY_TIMEOUT_API
```

Comment:
@carlescufi Some things will break as a result of this change, despite the changes in TI HAL.

One is the power management subsystem. I got a build error when building samples/boards/ti/cc13x2_cc26x2/system_off for cc1352r1_launchxl:

```
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:52:13: error: invalid operands to binary != (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   52 |  if ((ticks != K_FOREVER) && (ticks < pm_min_residency[0])) {
      |             ^~
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:63:14: error: invalid operands to binary == (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   63 |   if ((ticks == K_FOREVER) ||
      |              ^~
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:85:15: error: invalid operands to binary != (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   85 |     if (ticks != K_FOREVER) {
      |               ^~
[39/140] Building C object zephyr/CMak...vers/pinmux/pinmux_cc13xx_cc26xx.c.obj
ninja: build stopped: subcommand failed.
```
I think it goes beyond the above however. In subsys/power/power.c for instance, I see 
`        _sys_suspend(K_FOREVER);`
Yet _sys_suspend() is supposed to be called with ticks. It would be good to review the PM code to see what else needs to change, as much of it is conditionally compiled.

Other errors show up with the wifi driver as well when I compile samples/net/sockets/http_get for cc3220sf_launchxl:

```
[1/170] Preparing syscall dependency handling

[42/170] Building C object zephyr/CMak...fi/simplelink/simplelink_sockets.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj 
ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-2277-g12cbbeb6bf40 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/vwan/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/vwan/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vwan/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj   -c /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c: In function 'simplelink_poll':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:544:12: error: invalid operands to binary == (have 'int' and 'k_timeout_t' {aka 'struct <anonymous>'})
  544 |  if (msecs == K_FOREVER) {
      |            ^~
[45/170] Building C object zephyr/CMak...ivers/wifi/simplelink/simplelink.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj 
ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-2277-g12cbbeb6bf40 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/vwan/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/vwan/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vwan/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj   -c /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c: In function 'simplelink_scan_work_handler':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:114:48: error: incompatible type for argument 2 of 'k_delayed_work_submit'
  114 |   k_delayed_work_submit(&simplelink_data.work, delay);
      |                                                ^~~~~
      |                                                |
      |                                                s32_t {aka int}
In file included from ../../../../../include/zephyr.h:18,
                 from /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:10:
../../../../../include/kernel.h:3358:18: note: expected 'k_timeout_t' {aka 'struct <anonymous>'} but argument is of type 's32_t' {aka 'int'}
 3358 |      k_timeout_t delay)
      |      ~~~~~~~~~~~~^~~~~
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c: In function 'simplelink_mgmt_scan':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:148:48: error: incompatible type for argument 2 of 'k_delayed_work_submit'
  148 |   k_delayed_work_submit(&simplelink_data.work, delay);
      |                                                ^~~~~
      |                                                |
      |                                                s32_t {aka int}
In file included from ../../../../../include/zephyr.h:18,
                 from /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:10:
../../../../../include/kernel.h:3358:18: note: expected 'k_timeout_t' {aka 'struct <anonymous>'} but argument is of type 's32_t' {aka 'int'}
 3358 |      k_timeout_t delay)
      |      ~~~~~~~~~~~~^~~~~
[46/170] Building C object zephyr/CMak...r/subsys/net/lib/sockets/sockets.c.obj
ninja: build stopped: subcommand failed.
```

Reply:
Thanks for the comment. I will take a look tomorrow and fix as many as I can find. If there are additional ones we can fix them during the stabilization period for 2.3.